Util/ClassUtils.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Acl\Util;
+
+use Doctrine\Common\Util\ClassUtils as DoctrineClassUtils;
+
+/**
+ * Class related functionality for objects that
+ * might or might not be proxy objects at the moment.
+ *
+ * @see DoctrineClassUtils
+ *
+ * @author Benjamin Eberlei <kontakt@beberlei.de>
+ * @author Johannes Schmitt <schmittjoh@gmail.com>
+ */
+final class ClassUtils
+{
+    /**
+     * This class should not be instantiated.
+     */
+    private function __construct()
+    {
+    }
+
+    /**
+     * Gets the real class name of a class name that could be a proxy.
+     *
+     * @param string|object $object
+     *
+     * @return string
+     */
+    public static function getRealClass($object)
+    {
+        $class = is_object($object) ? get_class($object) : $object;
+
+        if (class_exists('Doctrine\Common\Util\ClassUtils')) {
+            return DoctrineClassUtils::getRealClass($class);
+        }
```

Comment:
you should also try using https://github.com/schmittjoh/cg-library/blob/master/src/CG/Core/ClassUtils.php when it is defined and Doctrine Common is not, to support proxies generated by this library (the naming convention is the same between both libraries precisely to allow having a simple way to find the real class for proxy classes)

and it is unfortunate that @Ocramius decided to use a different prefix in its own proxy-manager library, making it unsupported by our ``ClassUtils::getRealClass``